### PR TITLE
Update documentation

### LINKED ISSUE #80: Update user guide for bookmark
Update relevant sections of the user guide to include feature on bookmarking

### PR DESCRIPTION
**Pull request by vinceg4**
The following changes have been made in this PR:
- Update user guide to include documentation about bookmark feature
- Update developer guide's UML diagrams, the following 3 UML diagrams have been edited:
  - ArchitectureSequenceDiagram.puml
  - DeleteSequenceDiagram.puml
  - ModelClassDiagram.puml
 
Misc:
- Enable assertions (following tp task requirement)

Closes #80 #81 #83

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -69,4 +69,8 @@ shadowJar {
     archiveFileName = 'techconnect.jar'
 }
 
+run  {
+    enableAssertions = true
```
Comment (LingZiyann) [inline build.gradle:73]: Would like to see some assert statements being used

File: docs/UserGuide.md [hunk 2]
```diff
@@ -4,9 +4,9 @@
   pageNav: 3
 ---
 
-# AB-3 User Guide
+# TechConnect User Guide
 
-AddressBook Level 3 (AB3) is a **desktop app for managing contacts, optimized for use via a  Line Interface** (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, AB3 can get your contact management tasks done faster than traditional GUI apps.
+TechConnect is a **desktop app for managing contacts, optimized for use via a  Line Interface** (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, AB3 can get your contact management tasks done faster than traditional GUI apps.
```
Comment (LingZiyann) [inline docs/UserGuide.md:9]: Description talks about managing contacts but should be internship applications. Can change when you have the time

File: docs/UserGuide.md [hunk 3]
```diff
@@ -155,6 +158,32 @@ Shows a list of all companies in the address book.
...
+* Removes a bookmark from a bookmarked company at the specified `INDEX`.
+* The index refers to the index number shown in the displayed company list.
+* The index **must be a positive integer** 1, 2, 3, …​
+
+### Listing all bookmarked companies: `bmlist`
+
+Shows a list of all the bookmarked companies in the address book.
+
+Format: `bmlist`
+
```
Comment (LingZiyann) [inline docs/UserGuide.md:186]: Looks good

File: docs/diagrams/ModelClassDiagram.puml [hunk 4]
```diff
@@ -12,13 +12,15 @@ Class AddressBook
 Class ModelManager
 Class UserPrefs
 
-Class UniquePersonList
-Class Person
+Class UniqueCompanyList
```
Comment (LingZiyann) [inline docs/diagrams/ModelClassDiagram.puml:15]: Great changes!

File: docs/diagrams/ModelClassDiagram.puml [hunk 5]
```diff
@@ -35,20 +37,24 @@ ModelManager -left-> "1" AddressBook
...
-Person *--> "*" Tag
+AddressBook *--> "1" UniqueCompanyList
+UniqueCompanyList --> "~* all" Company
+Company *--> Name
+Company *--> Phone
+Company *--> Email
+Company *--> Bookmark
+Company *--> CareerPageUrl
+Company *--> Address
+Company *--> "*" Tag
```
Comment (LingZiyann) [inline docs/diagrams/ModelClassDiagram.puml:48]: Nice changing of Person to Company!

**Review by LingZiyann — APPROVED** [review]
LGTM

File: docs/UserGuide.md [hunk 6]
```diff
@@ -155,6 +158,32 @@ Shows a list of all companies in the address book.
...
+Removes a company from the list of bookmarked companies in the address book.
+
+Format: `removebm INDEX`
+
+* Removes a bookmark from a bookmarked company at the specified `INDEX`.
+* The index refers to the index number shown in the displayed company list.
+* The index **must be a positive integer** 1, 2, 3, …​
+
+### Listing all bookmarked companies: `bmlist`
+
```
Comment (tckeong) [inline docs/UserGuide.md:182]: LGTM!

**Review by tckeong — COMMENTED** [review]
LGTM overall, especially the change of the UML diagrams.

**Pull request merged**